category_encoders/target_encoder.py

```diff
@@ -113,34 +127,52 @@ def _fit(self, X, y, **kwargs):
 
     def fit_target_encoding(self, X, y):
         mapping = {}
+        prior = self._mean = y.mean()
 
         for switch in self.ordinal_encoder.category_mapping:
             col = switch.get('col')
-            values = switch.get('mapping')
+            if 'HEIR_' not in col:
+                values = switch.get('mapping')
 
-            prior = self._mean = y.mean()
+                scalar = prior
+                if self.heirarchy and col in self.heirarchy:
+                    col_heir = 'HEIR_'+col
```

Comment:
this also seems to work for single-level hierarchies only, right? So at some point we'd need to raise an exception (probably when parsing the hierarchy map)

Reply:
Agreed.